### DISCUSSION TITLE
训练ConvNext版本的YOLOv7和YOLOX

### DISCUSSION THREAD

**Discussion by lucasjinreal** [jinfagang/yolov7#46] [Ideas]
这里记录社区贡献者训练过程。

## 配置

主要是针对Convnext来做性能探索，由于Convnext同时兼备卷积的效率，以及transformer的性能，非常适合探索一波它能否超越我们常用的baseline resnet50，同时是否又能够在效率上超过卷积，需要深入试验探索。

该实验使用配置：https://github.com/jinfagang/yolov7/blob/main/configs/coco/yolox/yolox_convnext.yaml

训练任务启动脚本：

```
python train_net.py --config-file configs/coco/yolox/yolox_convnext.yaml --num-gpus 2
```

需要解决定位的问题：

1. ConvNext输出的是4层featuremap，需要在配置文件内改为3层

**Comment by PeterSmith1** [comment]
![image](https://user-images.githubusercontent.com/54202525/170907872-881ce713-2ee3-4475-8359-8f85aeeed4ed.png)
您好，训练时出现这个错误

**Comment by lucasjinreal** [comment]
@PeterSmith1  可以pull一下。应该可以训练了。

第一步我们先测试从 0, 1, 2三层拿FPN feature，可能不一定work，先训练一波看看。后面试一下从 1, 2, 3 拿feature.

可以继续再这里同步训练中间结果，如果情况不对就再调整
